node/runtime/src/lib.rs

```diff
@@ -773,14 +773,29 @@ impl Runtime {
             }
         };
         if needs_removal {
-            state_update.rollback();
-            state_update.delete(&callback_id_to_bytes(&callback_res.info.id));
-            state_update.commit();
+            if receipts.is_err() {
+                // On error, we rollback previous changes and then commit the deletion
+                state_update.rollback();
+                state_update.delete(&callback_id_to_bytes(&callback_res.info.id));
+                state_update.commit();
+            } else {
+                state_update.delete(&callback_id_to_bytes(&callback_res.info.id));
+            }
+            
             set(
```

Comment:
should we put the receiver acc update into the `else` clause in this case?

Reply:
Good point. Fixed in 05e7f8a